### DISCUSSION TITLE
Dispatching multiple request at controller level

### DISCUSSION THREAD

**Discussion by AsuraKev** [LuckyPennySoftware/MediatR#1083] [Q&A]
Hi there, I have a question regarding the usage of dispatching mediator request. I know that its not recommended to dispatch a request within a handler, but how about calling mulitple handlers at controller level. thanks!

e.g 

class Controller {

method() {
  var result1 = await mediator.Send();
  var result2 = await mediator.Send()

  return new  {
    a = result1,
    b = result2
  }
}

}

**Comment by zachpainter77** [comment]
If that's what you need to do then I don't see an issue with it.. you could also create a specific command that builds the dto with one request.. 

```csharp
var result = await _mediator.Send(new CreateMyDtoRequest());
return result;
```
